### PR TITLE
{PrivateDns} Track 2 Migration

### PR DESCRIPTION
**Pull request by rajitaaa**
**Description**

Changes made for track 2 migration:

Model: 
`RecordSetPaged` --> `RecordSetListResult`
Methods: 
`delete `---> `begin_delete`
`update` ---> `begin_update`
`create_or_update` --->` begin_create_or_update`
Exception: 
`CloudError` ---> `HttpResponseError` and `ResourceNotFoundError`
Removed unnecessary subitems for:
`operations_tmpl='azure.mgmt.privatedns.operations#PrivateZonesOperations.{}`
`operations_tmpl='azure.mgmt.privatedns.operations#VirtualNetworkLinksOperations.{}`
`operations_tmpl='azure.mgmt.privatedns.operations#RecordSetsOperations.{}`

Updated privatedns package to https://pypi.org/project/azure-mgmt-privatedns/1.0.0/

**Testing Guide**

**History Notes**

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
PrivateDns

**Comment by jsntcy** [comment]
@rajitaaa, please resolve the conflict.

File: scripts/sdk_process/patch_models.py [hunk 1]
```diff
@@ -234,7 +234,8 @@ def find_autorest_generated_folder(module_prefix="azure.mgmt"):
         'azure.mgmt.rdbms'
         'azure.mgmt.loganalytics',
         'azure.mgmt.web',
-        'azure.mgmt.cosmosdb'
+        'azure.mgmt.cosmosdb',
+       'azure.mgmt.privatedns'
```
Comment (jsntcy) [inline scripts/sdk_process/patch_models.py:238]:
```suggestion
        'azure.mgmt.privatedns'
```

**Review by jsntcy — APPROVED** [review]

**Review by fengzhou-msft — APPROVED** [review]
changed in `patch_models.py` LGTM.

**Pull request merged**